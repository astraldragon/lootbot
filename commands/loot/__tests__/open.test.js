jest.mock("discord.js-commando");
jest.mock("chance");
jest.mock("../../../database");

const chance = require("chance");
chance.mockReturnValue({ weighted: jest.fn() });

const { Command } = require("discord.js-commando");

const OpenCommand = require("../open");
const database = require("../../../database");

jest.useFakeTimers();

describe("commands/loot/open", () => {
  beforeEach(() => {
    chance().weighted = jest.fn();
  });

  it("exists", () => {
    expect(OpenCommand).toBeDefined();
  });

  it("configures properly", () => {
    const client = jest.fn("client");
    const command = new OpenCommand(client);
    expect(Command).toMatchSnapshot();
  });

  xit("shows the result of the draw", () => {
    const user = { id: 1 };
    const command = new OpenCommand(jest.fn());
    const loot = [{ name: "Test", weight: 100 }];
    const msg = {
      say: jest.fn().mockReturnValue(Promise.resolve()),
      guild: { id: 1 }
    };

    chance().weighted.mockReturnValue(loot[0]);
    database.list = jest.fn().mockImplementation(() => Promise.resolve(loot));

    const result = command.run(msg, { user });

    jest.runAllTimers();
    expect(msg.say).toHaveBeenCalled();
  });

  xit("shows the no loot message", async () => {
    const loot = [];
    const msg = { say: jest.fn() };
    const command = new OpenCommand(jest.fn());

    database.list = jest.fn().mockImplementation(() => Promise.resolve(loot));
    const result = await command.run(msg);

    expect(chance().weighted).not.toHaveBeenCalled();
    expect(msg.say).toHaveBeenCalledWith("No loot in the lootbox.");
  });
});
