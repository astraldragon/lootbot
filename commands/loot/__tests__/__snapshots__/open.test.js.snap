// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`commands/loot/open configures properly 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      [MockFunction],
      Object {
        "args": Array [
          Object {
            "key": "user",
            "prompt": "Which user would you like to open a lootbox for?",
            "type": "member",
          },
        ],
        "description": "Opens a lootbox",
        "examples": Array [
          "open",
        ],
        "group": "loot",
        "guildOnly": true,
        "memberName": "reply",
        "name": "open",
        "userPermissions": Array [
          "MANAGE_CHANNELS",
        ],
      },
    ],
  ],
}
`;
